genestack_client/settings/genestack_user.py

```diff
@@ -88,26 +88,43 @@ def __interactive_login(self, connection):
 
         email = self.email
         message = 'Connecting to %s' % self.host
+
+        choose_by_token = 'by token'
+        choose_by_email = 'by email and password'
+        choose_anonymously = 'anonymously'
+
+        choice = interactive_select([choose_by_token, choose_by_email, choose_anonymously],
+                                    'How do you want to login')
+
+        if choice == choose_anonymously:
+            return
+
         while True:
             if message:
                 print(message)
-            if email and '@' in email:
-                email = raw_input('e-mail [%s]: ' % email).strip() or email
+
+            if choice == choose_by_email:
+                if email and '@' in email:
+                    email = raw_input('e-mail [%s]: ' % email).strip() or email
+                else:
+                    email = raw_input('e-mail: ').strip() or email
+                password = getpass('password for %s: ' % email)
+                try:
+                    connection.login(email, password)
+                    self.email = email
+                    self.password = password
+                    return
+                except GenestackAuthenticationException:
+                    message = ('Your username or password was incorrect for %s. '
```

Comment:
`'Your username and password have been rejected by %s. '

Can we afford to change text message?  I hope so.